tests/e2e/vms/xsvm.go

```diff
@@ -196,32 +201,40 @@ var _ = ginkgo.Describe("[XSVM]", ginkgo.Label("xsvm"), func() {
 		node, err := network.GetNode(nodeID)
 		require.NoError(err)
 
-		uri := strings.TrimPrefix(node.URI, "http://")
-		chainID := network.GetSubnet(subnetAName).Chains[0].ChainID
-		client, conn, err := api.NewPingClient(
-			uri,
-			chainID,
-			grpc.WithTransportCredentials(insecure.NewCredentials()),
-		)
-		require.NoError(err)
-		ginkgo.DeferCleanup(func() {
-			require.NoError(conn.Close())
-		})
+		httpClient := &http.Client{
+			Transport: &http2.Transport{
+				AllowHTTP: true,
+				DialTLSContext: func(_ context.Context, network, addr string, _ *tls.Config) (net.Conn, error) {
+					// Skip TLS to use h2c
+					return net.Dial(network, addr)
+				},
+			},
+		}
+
+		client := xsvmconnect.NewPingClient(httpClient, node.URI)
 
 		tc.By("serving unary rpc")
 		msg := "foobar"
-		reply, err := client.Ping(tc.DefaultContext(), &xsvm.PingRequest{
-			Message: msg,
-		})
+
+		request := &connect.Request[xsvm.PingRequest]{
+			Msg: &xsvm.PingRequest{
+				Message: msg,
+			},
+		}
+
+		chainID := network.GetSubnet(subnetAName).Chains[0].ChainID
+		request.Header().Set(server.HTTPHeaderRoute, chainID.String())
+
+		reply, err := client.Ping(tc.DefaultContext(), request)
 		require.NoError(err)
-		require.Equal(msg, reply.Message)
+		require.Equal(msg, reply.Msg.Message)
 
 		tc.By("serving bidirectional streaming rpc")
-		stream, err := client.StreamPing(tc.DefaultContext())
-		require.NoError(err)
 
+		stream := client.StreamPing(tc.DefaultContext())
+		stream.RequestHeader().Set(server.HTTPHeaderRoute, chainID.String())
```

Comment:
This initialization seems pretty clunky.

Can we implement a helper (similarly to how we did with the grpc library)?

Specifically, I think we should be able to have a single helper that takes in the `node.URI` and `chainID` and it should return a `xsvmconnect.PingClient` which automatically sets the `server.HTTPHeaderRoute` by using the `connect.Interceptors` that we can pass into `xsvmconnect.NewPingClient`.